src/hdmf/validate/validator.py

```diff
@@ -164,7 +164,7 @@ def get_type(data, builder_dtype=None):
                 # Empty array
                 else:
                     # Empty string array
-                    if data.dtype.metadata["vlen"] == str:
+                    if isinstance(data.dtype.metadata["vlen"], str):
```

Comment:
```suggestion
                    if data.dtype.metadata["vlen"] is str:
```

I think the original code here was checking if it is the str class itself